backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java

```diff
@@ -64,14 +66,25 @@ public class NotificationResource {
     @Inject
     BehaviorGroupRepository behaviorGroupRepository;
 
+    @Inject
+    BehaviorGroupRepositoryOrgId behaviorGroupRepositoryOrgId;
```

Comment:
Why is this one needed? Can't the `BehaviorGroupRepository` use the configuration to decide what field to use?

Reply:
I started with duplicating the methods in the BehaviorGroupRepository first. But this end in a god class where it was horrible to find the methods to be deleted after the migration has been finished. That's why I decided to split it, so we have to remove the old BehaviorGroupRepository which deals with accountIds only. 

Methods above the call hierarchy I am having a switch for accountId -> BehaviorGroupRepository or BehaviorGroupRepositoryOrgId -> orgId.

Reply:
The NotificationResource has got both, the BehaviorGroupRepository and BehaviorGroupRepositoryOrgId and decides what to use. 

When we switch to orgId, we have to remove the if/else blocks in this class, delete the BehaviorGroupRepository class and use BehaviorGroupRepositoryOrgId everywhere. And rename BehaviorGroupRepositoryOrgId to BehaviorGroupRepository :)